libraries/lib-audio-io/AudioIO.cpp

```diff
@@ -2110,7 +2110,7 @@ bool AudioIO::ProcessPlaybackSlices(
 
          if(len > 0)
          {
-            for(unsigned i = 0; i < seq->NChannels(); ++i)
+            for(unsigned i = 0, cnt = std::min(seq->NChannels(), mNumPlaybackChannels); i < cnt; ++i)
```

Comment:
> Does it then crash?

I'd say it should, but didn't check myself

> AFAICS, if we were doing things properly further down the call stack and mixed left and right rather than only keeping the left channel

```cpp
for(unsigned n = 0, cnt = std::min(numChannels, mNumPlaybackChannels); n < cnt; ++n)
{
   const auto gain = seq->GetChannelGain(n);
   for(unsigned i = 0; i < samplesAvailable; ++i)
      mMasterBuffers[n][i] += mProcessingBuffers[bufferIndex + n][i] * gain;
}
```

`Keep-only-left-channel` logic implemented in per-track effects, when source track is mono and playback is stereo, that could be clearly observed with spatial effects